internal/cmd/oohelperd/internal/nwcth/measure.go

```diff
@@ -0,0 +1,281 @@
+package nwcth
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"net"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"sync"
+
+	"github.com/lucas-clemente/quic-go"
+	"github.com/ooni/probe-cli/v3/internal/engine/experiment/nwebconnectivity"
+)
+
+type (
+	// CtrlRequest is the request sent to the test helper
+	CtrlRequest = nwebconnectivity.ControlRequest
+
+	// CtrlResponse is the response from the test helper
+	CtrlResponse = nwebconnectivity.ControlResponse
+
+	CtrlURLMeasurement = nwebconnectivity.ControlURL
+
+	CtrlEndpointMeasurement = nwebconnectivity.ControlEndpoint
+
+	CtrlHTTPMeasurement = nwebconnectivity.ControlHTTP
+
+	CtrlH3Measurement = nwebconnectivity.ControlH3
+
+	CtrlTLSMeasurement = nwebconnectivity.ControlTLSHandshake
+
+	CtrlHTTPRequest = nwebconnectivity.ControlHTTPRequest
+)
+
+// NextLocationInfo contains the redirected location as well as the request object which forwards most headers of the initial request.
+// This forwarded request is generated by the http.Client and
+type NextLocationInfo struct {
+	jar             *cookiejar.Jar `json:"-"`
+	location        string         `json:"-"`
+	httpRedirectReq *http.Request  `json:"-"`
+}
+
+type MeasureURLResult struct {
```

Comment:
If possible use the same naming of the spec (we can discuss changing names in the spec though).

Also, please, document public structures.